tmt/steps/provision/__init__.py

```diff
@@ -3482,3 +3511,183 @@ def _run_action_phases(phases: list[Action]) -> tuple[list[ActionTask], list[Act
         self.summary()
         self.status('done')
         self.save()
+
+
+class AnsibleInventory:
+    """
+    Generate Ansible inventory files from provisioned guests.
+
+    Creates Ansible inventory file that can be used with playbooks
+    to manage provisioned guests. Supports custom layouts and automatically
+    configures host variables based on guest properties.
+    """
+
+    def __init__(self, plan: 'tmt.Plan') -> None:
+        """
+        Initialize the Ansible inventory handler.
+
+        :param plan: the plan containing provisioned guests and configuration.
+        """
+        self._logger = plan._logger
+        self._plan = plan
+
+    def _load_layout(self, layout_path: Optional[str] = None) -> dict[str, Any]:
+        """
+        Load inventory layout from file or use default.
+
+        :param layout_path: path to a custom layout file, relative to the FMF file.
+        # TODO check layout path logic
+        :returns: dictionary representing the inventory layout structure.
+        """
+        if layout_path:
+            try:
+                # Resolve path relative to the directory containing the FMF file
+                if self._plan.node.sources:
+                    # Use the directory of the first source (where the FMF file is)
+                    fmf_dir = Path(self._plan.node.sources[0]).parent
+                    resolved_path = fmf_dir / layout_path
+                else:
+                    # Fallback to current working directory if no sources
+                    resolved_path = Path(layout_path)
+                return tmt.utils.yaml_to_dict(resolved_path)
+            except tmt.utils.GeneralError as exc:
+                self._logger.warning(f"Failed to load inventory layout: {exc}")
+                return self._default_layout()
+        return self._default_layout()
+
+    def _default_layout(self) -> dict[str, Any]:
+        """
+        Create default inventory layout.
+
+        :returns: basic inventory structure with 'all' and 'ungrouped' groups.
+        """
+        return {'all': {'children': {'ungrouped': {}}}}
+
+    def _get_guest_ansible_config(self, guest: Guest) -> Optional[dict[str, Any]]:
```

Comment:
Yeah, we will have this propagated from phase to `Guest` instances, just as `role` or any other attribute of a `provision` phase. This method will not be needed, simple `guest.ansible` will work. See `Guest.hardware` and `GuestData.hardware`, it's the same story.